js/mexc.js

```diff
@@ -1142,18 +1142,14 @@ module.exports = class mexc extends Exchange {
 
     async fetchBalance (params = {}) {
         await this.loadMarkets ();
-        const defaultType = this.safeString2 (this.options, 'fetchBalance', 'defaultType', 'spot');
-        const type = this.safeString (params, 'type', defaultType);
-        const spot = (type === 'spot');
-        const swap = (type === 'swap');
-        let method = undefined;
-        if (spot) {
-            method = 'spotPrivateGetAccountInfo';
-        } else if (swap) {
-            method = 'contractPrivateGetAccountAssets';
-        }
-        const query = this.omit (params, 'type');
-        const response = await this[method] (query);
+        let marketType = undefined;
+        [ marketType, params ] = this.handleMarketTypeAndParams ('fetchBalance', undefined, params);
+        const method = this.getSupportedMapping (marketType, {
+            'spot': 'spotPrivateGetAccountInfo',
+            'swap': 'contractPrivateGetAccountAssets',
+        });
+        const spot = (marketType === 'spot');
+        const response = await this[method] (this.extend (params));
```

Comment:
Here, `this.extend` takes at least two params, the object to extend, and the object to extend with. We don't need `this.extend`, if we're sending just the `params` to the exchange.